### ISSUE TITLE
Template Haskell Link Error

### ISSUE THREAD

**Issue by cschneid** [ghcjs/ghcjs#330]
See example repo: https://bitbucket.org/cschneid/grocerygopher2 -- Its a bit difficult to build though, hopefully the code can be of some help.

Basically, when I run `configure -fclient --ghcjs` then `cabal build` I get:

```
Warning: Could not list sources of the add-source dependency 'blaze-react'.
Skipping the timestamp check.
Building grocery-0.1.0.0...
Preprocessing library grocery-0.1.0.0...
[10 of 13] Compiling Grocery.Types.User ( src/shared/Grocery/Types/User.hs, dist/build/Grocery/Types/User.js_o )
Linking Template Haskell (Data.Aeson.Day,Data.Aeson.UUID,Data.Time.DayOfWeek,Grocery.ListHelpers,Grocery.Types,Grocery.Types.Food,Grocery.Types.UserPassword,Network.Http.StatusHelpers,ThRunner1)

<no location info>:
    exported function not found: Fun {funPackage = Package {unPackage = "groce_AtInTsMKVozEZak4rjy0ln"}, funModule = "Data.Aeson.Day", funSymbol = "h$$grocezuAtInTsMKVozzEZZak4rjy0lnZCDataziAesonziDayzia1_FBcfCV"}
[11 of 13] Compiling Grocery.Types.Recipe ( src/shared/Grocery/Types/Recipe.hs, dist/build/Grocery/Types/Recipe.js_o )
Linking Template Haskell (Data.Aeson.Day,Data.Aeson.UUID,Data.Time.DayOfWeek,Grocery.ListHelpers,Grocery.Types,Grocery.Types.Food,Network.Http.StatusHelpers,ThRunner2)

<no location info>:
    exported function not found: Fun {funPackage = Package {unPackage = "groce_AtInTsMKVozEZak4rjy0ln"}, funModule = "Data.Aeson.Day", funSymbol = "h$$grocezuAtInTsMKVozzEZZak4rjy0lnZCDataziAesonziDayzia1_FBcfCV"}
```

Note that the package it claims is missing is in my repo (the library section, exported).

Running ghc 7.10.1, ghcjs installed as of today.

**Comment by cschneid** [comment]
I wonder if this has something to do with the fact that the `Data.Aeson.Day` module is simply to export an orphan instance? 

https://bitbucket.org/cschneid/grocerygopher2/src/b467e3b7f01f3750023f26f98d193164e5fe59b0/src/shared/Data/Aeson/Day.hs?at=master

**Comment by luite** [comment]
Hmm, I'll try to reproduce this. 7.10 has some improvements wrt package and dependency loading for TH, it might be part of that.

More likely though it's the instance being partially inlined, and the code depending on an internal symbol name of the module that is not added to the export list of the corresponding object file.

**Comment by cschneid** [comment]
I did see this on 7.8 as well, so I think that this isn't related to 7.10 itself.

**Comment by cschneid** [comment]
Another very odd data point - it appears that `cabal install -fclient --ghcjs` finishes, but the `cabal configure -fclient --ghcjs; cabal build` fails with the above error.

**Comment by luite** [comment]
probably fixed in a73f65a9d9d8639cd23707edc4fdc06fb77c092d , can you verify?

**Comment by luite** [comment]
please reopen if you still see this
